pkg/crinit/common/common.go

```diff
@@ -451,3 +486,62 @@ func WaitSrvHealthy(cmdOut io.Writer, crClientset client.Interface) error {
 	}
 	return err
 }
+
+// DeleteKubeconfigEntry handles updating the kubeconfig to remove the cluster
+// registry entry that was previously added when the cluster registry was
+// initialized.
+func DeleteKubeconfigEntry(cmdOut io.Writer, pathOptions *clientcmd.PathOptions,
+	name, kubeconfig string, dryRun bool) error {
+
+	fmt.Fprintf(cmdOut, "Delete kubeconfig entry %s...", name)
+	glog.V(4).Infof("Delete kubeconfig entry %s", name)
+
+	err := util.DeleteKubeconfigEntry(cmdOut, pathOptions, name, kubeconfig, dryRun)
+
+	if err != nil {
+		glog.V(4).Infof("Failed to delete kubeconfig entry %s: %v", name, err)
+		return err
+	}
+
+	fmt.Fprintln(cmdOut, " done")
+	glog.V(4).Info("Successfully deleted kubeconfig entry")
+	return nil
+}
+
+// WaitForClusterRegistryDeletion waits a certain amount of time for the namespace
+// requested to be deleted.
+func WaitForClusterRegistryDeletion(cmdOut io.Writer, clientset client.Interface,
+	namespace string, dryRun bool) error {
+
+	if dryRun {
+		_, err := fmt.Fprintln(cmdOut, "Cluster registry can be deleted (dry run)")
+		glog.V(4).Infof("Cluster registry can be deleted (dry run)")
+		return err
+	}
+
+	fmt.Fprint(cmdOut, "Waiting for the cluster registry to be deleted...")
+	glog.V(4).Info("Waiting for the cluster registry to be deleted")
+
+	var getErr error
+	err := wait.PollImmediate(waitInterval, waitTimeout, func() (bool, error) {
```

Comment:
Namespace deletion can take a very long time in my experience. We may need a longer timeout here, or some other way of indicating in the returned error that this is not necessarily unexpected.

Perhaps this command should continue and report some pseudo-success?